src/VecSim/spaces/IP_space.cpp

```diff
@@ -6,121 +6,85 @@
 
 #include "VecSim/spaces/IP_space.h"
 #include "VecSim/spaces/IP/IP.h"
-#include "VecSim/spaces/IP/IP_AVX512DQ.h"
+#if defined(__x86_64__)
 #include "VecSim/spaces/IP/IP_AVX512.h"
 #include "VecSim/spaces/IP/IP_AVX.h"
 #include "VecSim/spaces/IP/IP_SSE.h"
```

Comment:
Are we sure it is dafe to include higher optimization in older platforms? (AVX in SSE for example)